src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.cpp

```diff
@@ -0,0 +1,199 @@
+using namespace chip::app::Clusters::LaundryWasherControls;
+using namespace chip::app::Clusters::LaundryWasherControls::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+static constexpr size_t kLaundryWasherControlsDelegateTableSize =
+    EMBER_AF_LAUNDRY_WASHER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+// -----------------------------------------------------------------------------
+// Delegate Implementation
+//
+namespace {
+Delegate * gDelegateTable[kLaundryWasherControlsDelegateTableSize] = { nullptr };
+}
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace LaundryWasherControls {
+
+void SetDefaultDelegate(EndpointId endpoint, Delegate * delegate)
+{
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, LaundryWasherControls::Id,
+                                                       EMBER_AF_LAUNDRY_WASHER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT);
+    // if endpoint is found
+    if (ep < kLaundryWasherControlsDelegateTableSize)
+    {
+        gDelegateTable[ep] = delegate;
+    }
+    else
+    {
+    }
+}
+} // namespace LaundryWasherControls
+} // namespace Clusters
+} // namespace app
+} // namespace chip
+
+namespace {
+Delegate * GetDelegate(EndpointId endpoint)
+{
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, LaundryWasherControls::Id,
+                                                       EMBER_AF_LAUNDRY_WASHER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return (ep >= kLaundryWasherControlsDelegateTableSize ? nullptr : gDelegateTable[ep]);
+}
+
+bool isDelegateNull(Delegate * delegate, EndpointId endpoint)
+{
+    if (delegate == nullptr)
+    {
+        ChipLogProgress(Zcl, "Laundry Washer Control has no delegate set for endpoint:%u", endpoint);
+        return true;
+    }
+    return false;
+}
+
+class LaundryWasherControlsAttrAccess : public AttributeAccessInterface
+{
+public:
+    LaundryWasherControlsAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), LaundryWasherControls::Id) {}
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR ReadSpinSpeeds(AttributeValueEncoder & aEncoder, Delegate * delegate);
+    CHIP_ERROR ReadSpinSpeedCurrent(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadNumberOfRinses(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadSupportedRinses(AttributeValueEncoder & aEncoder, Delegate * delegate);
+};
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSpinSpeeds(AttributeValueEncoder & aEncoder, Delegate * delegate)
+{
+    return delegate->HandleGetSpinSpeedsList(aEncoder);
+}
+
+CHIP_ERROR LaundryWasherControlsAttrAccess::ReadSpinSpeedCurrent(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    DataModel::Nullable<uint8_t> speedSetting;
+    SpinSpeedCurrent::Get(endpoint, speedSetting);
```

Comment:
This doesn't make sense.  If this value is being stored in the attribute store, why do we need to have this code at all?  Just let the generic machinery read it from the store.

But then we need a cluster API for the application to set the value, without prodding the attribute store directly.... (the cluster API could store in the attr store, if that's where we want to keep it).

The other option is that we don't store this in the attribute store at all and just ask the delegate for this value.

Reply:
I will remove them.